File: MiniDOMWebLibs/MiniDOM/Collegamenti/Collegamenti.vb
Imports minidom
Imports minidom.Sistema
Imports minidom.Databases

Partial Class WebSite
    ''' <summary>
    ''' Gestione del collegamenti in prima pagina
    ''' </summary>
    ''' <remarks></remarks>
    Public NotInheritable Class CCollegamentiClass
        Inherits CModulesClass(Of CCollegamento)

        Friend Sub New()
            MyBase.New("modLinks", GetType(CCollegamentiCursor), -1)
        End Sub


        ''' <summary>
        ''' Restituisce i link visibili all'utente
        ''' </summary>
        ''' <param name="user"></param>
        ''' <returns></returns>
        ''' <remarks></remarks>
        Public Function GetUserLinks(ByVal user As CUser) As CCollection(Of CCollegamento)
            If (user Is Nothing) Then Throw New ArgumentNullException("user")

            Dim ret As New CCollection(Of CCollegamento)

            If (GetID(user) = 0) Then Return ret

            Dim items As CCollection(Of CCollegamento) = Me.LoadAll
            For Each item As CCollegamento In items
                If (item.Stato = ObjectStatus.OBJECT_VALID AndAlso item.IsVisibleToUser(user)) Then ret.Add(item)
            Next
            ret.Sort()
            Return ret
        End Function

    End Class

    Private Shared m_Collegamenti As CCollegamentiClass = Nothing

    Public Shared ReadOnly Property Collegamenti As CCollegamentiClass
        Get
            If (m_Collegamenti Is Nothing) Then m_Collegamenti = New CCollegamentiClass
            Return m_Collegamenti
        End Get
    End Property



End Class

File: MiniDOMWebLibs1/MiniDOM/Collegamenti/Collegamenti.cs
using System;
using DMD;
using DMD.XML;
using minidom;
using minidom.Internals;
using static minidom.Sistema;


namespace minidom
{
    public partial class WebSite
    {
        /// <summary>
    /// Gestione del collegamenti in prima pagina
    /// </summary>
    /// <remarks></remarks>
        public sealed class CCollegamentiClass : Sistema.CModulesClass<CCollegamento>
        {
            internal CCollegamentiClass() : base("modLinks", typeof(CCollegamentiCursor), -1)
            {
            }


            /// <summary>
        /// Restituisce i link visibili all'utente
        /// </summary>
        /// <param name="user"></param>
        /// <returns></returns>
        /// <remarks></remarks>
            public CCollection<CCollegamento> GetUserLinks(Sistema.CUser user)
            {
                if (user is null)
                    throw new ArgumentNullException("user");
                var ret = new CCollection<CCollegamento>();
                if (Databases.GetID(user) == 0)
                    return ret;
                var items = LoadAll();
                foreach (CCollegamento item in items)
                {
                    if (item.Stato == Databases.ObjectStatus.OBJECT_VALID && item.IsVisibleToUser(user))
                        ret.Add(item);
                }

                ret.Sort();
                return ret;
            }
        }

        private static CCollegamentiClass m_Collegamenti = null;

        public static CCollegamentiClass Collegamenti
        {
            get
            {
                if (m_Collegamenti is null)
                    m_Collegamenti = new CCollegamentiClass();
                return m_Collegamenti;
            }
        }
    }
}